### PR TITLE
EVM-439 Custom Lint rule to check file permissions

### PR DESCRIPTION
**Pull request by goran-ethernal**
# Description

Added a new custom lint rule that checks if file permission `0777` is used in `os.*` package functions.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by vcastellm — APPROVED** [review]
LGTM if this is the most straightforward way to integrate ruleguard

**Review by igorcrevar — APPROVED** [review]

**Comment by goran-ethernal** [comment]
@vcastellm Only way without creating a custom plugin and having to build the plugin independently.

File: gorules/rules_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"testing"
+
+	"github.com/quasilyte/go-ruleguard/analyzer"
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func TestRules(t *testing.T) {
+	testdata := analysistest.TestData()
```
Comment (Stefan-Ethernal) [inline gorules/rules_test.go]: Can we include parallel execution?
```suggestion
func TestRules(t *testing.T) {
        t.Parallel()
	testdata := analysistest.TestData()
```

**Review by Stefan-Ethernal — APPROVED** [review]

File: gorules/rules_test.go (hunk 1 above)
Comment (goran-ethernal) [reply gorules/rules_test.go]: Sure: https://github.com/0xPolygon/polygon-edge/pull/1271/commits/5798203c6b32c0899999e92d84fd0bddb3940899

**Pull request merged**